script/bootstrap-on-macOS-using-MacPorts.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+#
+# bootstrap-on-macOS-using-MacPorts.sh
+#
+# Run this script with root privileges. I.e.:
+#
+# sudo ./script/bootstrap-on-macOS-using-MacPorts.sh
+#
+# Copyright (C) 2024 Stephen G. Tuggy and other Vega Strike contributors
+#
+# This file is part of Vega Strike.
+#
+# Vega Strike is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Vega Strike is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Vega Strike. If not, see <https://www.gnu.org/licenses/>.
+
+set -e
+
+port install python312 boost181@1.81.0_10+cmake_scripts+no_single+no_static+python312 gtk3 gtkglext libsdl2 libGLU freeglut xorg-server openal-soft cmake expat libjpeg-turbo libpng libvorbis ninja
+
+export CMAKE_PREFIX_PATH="/opt/local/"
+
+ln -s /opt/local/include/GL /opt/local/include/OpenGL
+ln -s /opt/local/include/GL /opt/local/include/GLUT
```

Comment:
This could be an issue for those without admin, perhaps use some CMake/build-time vars for some dynamic ability?

Reply:
Strictly speaking, I don't think these symbolic links are actually necessary. I will probably remove these two lines from this script. And the same with the other bootstrap-on-macOS script, the Homebrew version.

Reply:
Is it possible to setup `CMAKE_PREFIX_PATH` in `engine/CMakeLists.txt`? Is it possible to replace the links either with CMake configuration or with special-cased includes?

Reply:
Looking at this patch for example, the MacPorts paths match the non-Apple GL location:
https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/851/files#diff-2c914b3535341095b4fd784c9daa704b78cd59c235aab27669f1f1584d8b03fb

Reply:
We should be able to add some variables, and even detect and use homebrew on Mac. If we need to require sudo/root for now fine, but let's work towards not needing it.

Reply:
MacPorts requires sudo/root categorically. As for Homebrew, there's a different script for that.

I'd be happy to write a meta script that detects and uses one or the other. And yes, we could probably do without those links, now that you mention it. Can those items wait until a future PR though?

Reply:
Regarding `CMAKE_PREFIX_PATH`: The proper place to set that is probably in CMakePresets.json. I have the setting in my CMakeUserPresets.json; it would be good to get it into the main presets file as well.

Reply:
Okay. The items in this thread should now be resolved. I was able to refactor CMakeLists.txt to use find_file to find the appropriate OpenGL-related headers and their folder locations, then use that information to include the headers from every project file that uses them.

Note that this results in the `#include` lines looking like this:

```c++14
#include <glext.h>
```

instead of like this:

```c++14
#include <(Open)GL/glext.h>
```

I hope that this is an acceptable solution to the above items.

Reply:
Not including the directory name in OpenGL-related header `#include`s is a bit unconventional; that's my main concern with this approach. But I suppose it will be fine if I just document it properly with comments in the code.

I think I'll do that this morning, actually. And maybe get rid of a few unnecessary `#ifdef`s while I'm at it, where the two `#if` branches are literally the same. And after that, I'm done making changes to this PR. I promise LOL. (Unless you guys really think more changes are needed at that point.)

Reply:
@stephengtuggy thanks - let me know when you're done but I think you've got a good plan for 0.9.x from the comments above

Reply:
I believe I am done. I will go ahead and merge this thing!